lib/api/CAnomalyJob.cc

```diff
@@ -972,18 +972,18 @@ bool CAnomalyJob::restoreDetectorState(const model::CSearchKey& key,
     return true;
 }
 
-bool CAnomalyJob::persistState(core::CDataAdder& persister) {
+bool CAnomalyJob::persistState(core::CDataAdder& persister, const std::string& descriptionPrefix) {
     if (m_PeriodicPersister != nullptr) {
         // This will not happen if finalise() was called before persisting state
         if (m_PeriodicPersister->isBusy()) {
-            LOG_ERROR(<< "Cannot do final persistence of state - periodic "
-                         "persister still busy");
+            LOG_ERROR(<< "Cannot perform foreground persistence of state - periodic "
+                         "background persistence is still in progress");
```

Comment:
The message is correct for `finalise` because on graceful close we do a foreground persist just before the process exits.  We never do background persist during a graceful close because duplicating all the models at this point would just waste time.

But this ties in with my comment about calling `finalise` at times other than immediately before shutting down.  I don't think we should do that.